src/extension.ts

```diff
@@ -65,7 +65,13 @@ export function activate(context: vscode.ExtensionContext) {
             const isEmptySourceFile = (fileInfo.document.getText() == '' && fileInfo.ext.match(/^(?:c|cpp|C|cc)$/));
 
             if (isEmptyHeaderFile) {
-                const name = path.basename(fileInfo.fileName).replace('.', '_').replace("-", "_").concat("_");
+                let name = await vscode.window.showInputBox({ prompt: "Type header definer: ", placeHolder: "Leave empty to use filename as header..."})
+                if (name === undefined) {
+                    name = "";
+                } else if (name === '') {
+                    name = path.basename(fileInfo.fileName)
+                }
+                name = name.replace(/ /g, "_").replace('.', '_').replace(/-/g, "_").concat("_")
```

Comment:
Thinking about this and how I did this before and at other places in the code, I think I should have replaced by a whitelist rather than an incomplete blacklist.  
Something like:
```js
name.replace(/[^A-Za-z0-9]/g, "_")
```

You can tell me whether you want to do that change yourself or not.  
I will, soon enough, refactor quite a lot of code anyway (since it starts to become old code and I think it can be greatly improved, both in correctness and code style).

Reply:
I can do it on the next commit to have a better replacement on the filenames and you can take your time to refactor the code having a stable and well working version released.
So i will do all this new changes as soon as possible :rocket: